src/wallet/wallet.cpp

```diff
@@ -2420,10 +2420,11 @@ CWallet::Balance CWallet::GetBalance(const int min_depth, bool avoid_reuse) cons
     {
         auto locked_chain = chain().lock();
         LOCK(cs_wallet);
+        std::set<uint256> trustedParents;
```

Comment:
If you go with DoS protection, can't you go further and reuse the cached set of trusted parents between WalletTxToJSON call in `listtransactions` and `listsinceblock`, they are all after chain locks so shouldn't be an issue ?

Reply:
Possibly, but because it's through the interfaced call I didn't want to scope creep to include that. But it's maybe worth a follow up PR?

Reply:
Really IMO but if DoS protection is worth it, it should take precedence on creeping into interfaced call.